recipes/boost-ext-ut/all/conanfile.py

```diff
@@ -16,15 +18,15 @@ class UTConan(ConanFile):
 
     @property
     def _minimum_cpp_standard(self):
-        return 17
+        return 17 if self.settings.os == "Linux" else 20
```

Comment:
I would make this version dependent as well (instead of depending on the os).
It is the compiler that matters to c++ standard, not the os.
Linux needs c++20 as well, no?
I don't see written anything on the [cppreference site](https://en.cppreference.com/w/cpp/language/lambda) about it being os dependent.

Reply:
That's the thing, Boost UT compiles with C++17 just fine on Linux at the moment and their CI even uses C++ 17 in their CI for Linux, though not for Windows / macOS.
A macro check ensures the compiler supports generic lambdas.
The compiler checks make more sense, but I was concerned something would crop up like an issue with macOS's C++ standard library when used with LLVM Clang, but it's probably best to take that as it comes and account for it then.
I've changed this to use C++ 17 for Clang and GCC and C++ 20 for all other compilers.

Reply:
Indeed, it will probably be a (gcc,clang) vs apple-clang thing.
Nice fix/workaround you applied here.